pkg/storage/tsdb/block/fetcher.go

```diff
@@ -370,6 +391,12 @@ func (f *MetaFetcher) fetchMetadata(ctx context.Context, excludeMarkedForDeletio
 				return nil
 			}
 
+			// skip any blocks older than the fetcher's max lookback.
+			if id.Compare(minAllowedBlockID) == -1 {
```

Comment:
This would not work correctly if, for some reason, there was a default-constructed `ulid` among the input. Seems like slightly less chance of surprising behavior if the test is skipped altogether when lookback is disabled. (More thinking of accidentally changing default behavior if this experimental flag isn't enabled.)

```suggestion
			if f.maxLookback > 0 && id.Compare(minAllowedBlockID) == -1 {
```